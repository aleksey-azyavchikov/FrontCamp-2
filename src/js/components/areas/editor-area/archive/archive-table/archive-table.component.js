import { Component } from "../../../../../core/decorators/component.decorator";
import BaseComponent from "../../../../base.component";
import { ActionType } from "../../../../../reducers/action-type";
import { EditorMode } from "../../../../../core/enums/editor-mode.enum";
import { Boolean } from "tcomb";

@Component({
    selector: "fc-archive-table",
    template: require("./archive-table.component.html"),
    styles: require("./archive-table.component.scss"),
    children: []
})
export default class ArchiveTableComponent extends BaseComponent {
    constructor() {
        super();
    }
    
    initializeParamsHook() {
        this.articles = this.config.params.articles;
        this.selected = this.config.params.selected;
        this.mode = this.config.params.mode;
    }

    defineDomElementsHook() {
        let domElements = {
            rows: this.config.ref.querySelectorAll("tbody tr"),
            body: this.config.ref.querySelector("tbody"),
        };
        this.domElements = domElements;
    }

    bindHandlersHook() {
        this.bindEvent(this.domElements.body, "click", (event) => {
            let finded = this.articles.find(article => article._id === event.target.parentNode.id);
            this.selected = finded === this.selected ? null : finded;
            this.dispatch(this.selected);
            this.render();
        });
    }

    dispatch(selected) {
        if(!selected) {
            this.config.store.dispatch({ 
                type: ActionType.SetArchiveEditorMode,
                payload: EditorMode.None
            });
        }
        this.config.store.dispatch({ 
            type: ActionType.SelectArticle,
            payload: selected
        });
    }

    isEnabled() {
        console.log("mode", this.mode);
        let result = this.mode === EditorMode.None || this.mode === EditorMode.Delete;
        console.log("re", result);
        return result; 
    }
}