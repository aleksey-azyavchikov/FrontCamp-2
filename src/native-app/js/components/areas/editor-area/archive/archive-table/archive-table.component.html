<table class="table table-bordered table-hover">
    <thead class="thead-dark">
        <tr>
            <th scope="col">â„–</th>
            <th scope="col">Title</th>

            <th cmIf="this.isEnabled()" scope="col">Description</th>
            <th cmIf="this.isEnabled()" scope="col">Handle</th>

            <th scope="col">Link</th>
        </tr>
    </thead>
    <tbody cmIf="this.articles" cmFor="let article of this.articles">
        <tr id="{{this.article._id}}" class="article {{this.scope.selected && this.article._id === this.scope.selected._id ? 'selected-row' : ''}}">
            <th scope="row">{{this.index + 1}}</th>
            <td>{{this.article.title}}</td>

            <td cmIf="this.isEnabled()">{{this.article.description}}</td>
            <!-- <td cmIf="this.selected">...</td> -->
            
            <td cmIf="this.isEnabled()">{{this.article.publishedAt}}</td>
            <!-- <td cmIf="this.selected">...</td> -->

            <td><a href="{{this.article.url}}">Link</a></td>
        </tr>
    </tbody>
</table>