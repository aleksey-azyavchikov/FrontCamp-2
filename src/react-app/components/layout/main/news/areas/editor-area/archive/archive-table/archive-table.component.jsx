import { Guid } from "../../../../../../../../../core/guid";
import ArchiveTableRow from "./archive-table-row/archive-table-row.component";
import React from "react";
import "./archive-table.component.scss";

class ArchiveTable extends React.Component {

    render() {
        const headers = ["â„–", "Title", "Description", "Handle", "Link"].map((name, index) => <th key={index} scope="col">{name}</th>)
        const rows = this.props.articles.map((article, index) => <ArchiveTableRow key={Guid.create()} index={index + 1} article={article} />)

        return (
            <div class="container">
                <table class="table table-bordered table-hover archive-table--position">
                    <thead class="thead-dark">
                        <tr>
                            {headers}
                        </tr>
                    </thead>
                    <tbody>
                        {rows}
                    </tbody>
                </table>
            </div>
        );
    }
}

export default ArchiveTable;