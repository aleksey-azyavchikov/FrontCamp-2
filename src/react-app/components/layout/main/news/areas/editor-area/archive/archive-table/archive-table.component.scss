.archive-table--position {
    padding-top: 20px;
}