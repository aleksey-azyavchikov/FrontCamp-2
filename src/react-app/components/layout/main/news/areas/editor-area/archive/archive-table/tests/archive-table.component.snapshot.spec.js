import React from "react";
import configureStore from "redux-mock-store";
import { ArchiveTable } from "../archive-table.component";
import renderer from "react-test-renderer";

describe("Compoentn: Archive table:", () => {
    const initialState = { archiveState: { selected: null } };
    const mockStore = configureStore();
    let store;
    let props;
    beforeAll(() => {
        store = mockStore(initialState);
    })

    beforeEach(() => {
        props = { articles: [], store }
    })

    it("Display: initial", () => {
        const rendered = renderer.create(
            <ArchiveTable {...props}/>
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    })
}) 
