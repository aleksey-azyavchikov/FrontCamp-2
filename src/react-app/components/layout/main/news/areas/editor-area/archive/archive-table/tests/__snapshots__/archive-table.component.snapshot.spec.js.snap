// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compoentn: Archive table: Display: initial 1`] = `
<table
  className="table table-bordered table-hover"
>
  <thead
    className="thead-dark"
  >
    <tr>
      <th
        scope="col"
      >
        â„–
      </th>
      <th
        scope="col"
      >
        Title
      </th>
      <th
        scope="col"
      >
        Description
      </th>
      <th
        scope="col"
      >
        Date
      </th>
      <th
        scope="col"
      >
        Link
      </th>
    </tr>
  </thead>
  <tbody />
</table>
`;
